### ISSUE TITLE
PlanDefinition$apply should include a RequestGroup within the returned CarePlan

### ISSUE THREAD

**Issue by dehall** [DBCG/cqf-ruler#235]
According to the FHIR spec: 
> The result of [the PlanDefinition$apply] operation is a CarePlan resource with a single activity represented by a RequestGroup. The RequestGroup will have actions for each of the applicable actions in the plan based on evaluating the applicability condition in context.  

(ref: https://www.hl7.org/fhir/operation-plandefinition-apply.html , at the bottom)

However in cqf-ruler, when calling PlanDefinition$apply, the result is just a CarePlan with actions included directly as activities. A simple example with input and output, produced with the latest from `develop` is included below.

I notice that within the PlanDefinitionApplyProvider, there appear to be two code paths:
 - one for the actual PlanDefinition$apply operation: 
   - https://github.com/DBCG/cqf-ruler/blob/a3166f0b7d86c44da9e82ad599eeb5163bebd367/r4/src/main/java/org/opencds/cqf/r4/providers/PlanDefinitionApplyProvider.java#L80
 - and one "for library use" called `resolveCdsHooksPlanDefinition()`
   - https://github.com/DBCG/cqf-ruler/blob/a3166f0b7d86c44da9e82ad599eeb5163bebd367/r4/src/main/java/org/opencds/cqf/r4/providers/PlanDefinitionApplyProvider.java#L255
   - this one does appear to return a RequestGroup, I haven't looked closely into what else might be different

Is there any active work being done on the PlanDefinition to have it return a RequestGroup, or harmonize between the two execution flows? I did see some work done on the Evaluation refactor PR (#189) but that branch still seems to not produce the RequestGroup. 

My team is looking at using cqf-ruler to evaluate FHIR CPGs  (http://build.fhir.org/ig/HL7/cqf-recommendations/index.html) so having that support would be very useful.

Thanks!

### Simple example
```
{
  "resourceType": "PlanDefinition",
  "id": "14849",
  "meta": {
    "versionId": "1",
    "lastUpdated": "2020-07-31T20:16:38.519+00:00"
  },
  "action": [
    {
      "id": "medication-action-1",
      "title": "Administer Medication 1",
      "definitionCanonical": "ActivityDefinition/14847"
    },
    {
      "id": "medication-action-2",
      "title": "Administer Medication 2",
      "definitionCanonical": "ActivityDefinition/14848"
    }
  ]
}

{
  "resourceType": "ActivityDefinition",
  "id": "14847",
  "kind": "MedicationRequest",
  "productCodeableConcept": {
    "text": "Medication 1"
  }
}

{
  "resourceType": "ActivityDefinition",
  "id": "14848",
  "kind": "MedicationRequest",
  "productCodeableConcept": {
    "text": "Medication 2"
  }
}
```

`GET cqf-ruler-r4/fhir/PlanDefinition/14849/$apply?subject=Patient/14549` -->
```
{
  "resourceType": "CarePlan",
  "contained": [
    {
      "resourceType": "MedicationRequest",
      "id": "94cdac61-49bf-44b1-9355-a44154b6e489",
      "intent": "order",
      "medicationCodeableConcept": {
        "text": "Medication 1"
      }
    },
    {
      "resourceType": "MedicationRequest",
      "id": "54234eb5-dff5-49ec-992b-cd2f56f66a54",
      "intent": "order",
      "medicationCodeableConcept": {
        "text": "Medication 2"
      }
    }
  ],
  "instantiatesCanonical": [
    "14849"
  ],
  "status": "draft",
  "activity": [
    {
      "reference": {
        "reference": "#94cdac61-49bf-44b1-9355-a44154b6e489"
      }
    },
    {
      "reference": {
        "reference": "#54234eb5-dff5-49ec-992b-cd2f56f66a54"
      }
    }
  ]
}
```

**Comment by JPercival** [comment]
Hi @dehall . Apologies for the delayed response. Yes, there was work done to harmonize the two code paths but it's now significantly out of date:

https://github.com/DBCG/cqf-ruler/pull/53

We will need to go back and re-work that. I don't have a timeframe at the moment due to the effort to support the current HL7 Connectathon preempting other things but I'll try to get it back on the radar.

**Comment by dehall** [comment]
@JPercival no worries on the delay. My team has actually implemented a couple changes on a fork of the repo, and we intend to open a PR to offer those changes upstream shortly. No pressure to accept but we want to be good open-source citizens.

**Comment by JPercival** [comment]
Resolved by #264
